carbonate_sync.py

```diff
@@ -0,0 +1,318 @@
+            '--rsh', ssh_cmd,
+            '--files-from=%s' % f.name,
+        ]
+        cmd += rsync_options
+        cmd += [
+            remote,
+            staging_dir,
+        ]
+        _run(cmd)
+
+
+def _make_staging_dir(base_dir, node):
+    """Creates the staging directory for this node."""
+    dirname = os.path.join(base_dir, node)
+    try:
+        os.makedirs(dirname)
+    except OSError as exc:
+        # Ignore already existing directories
+        if exc.errno != errno.EEXIST or not os.path.isdir(dirname):
+            raise
+    return dirname
+
+
+def _heal(args_and_metric):
+    """Heal whisper files.
+
+    This method will backfill data present in files in the
+    staging dir if not present in the local files for
+    points between 'start' and 'stop' (unix timestamps).
+    """
+    args, metric = args_and_metric
+    staging_dir, start, end = args
+    carbonate_sync.heal_metric(
+        os.path.join(staging_dir, metric),
+        os.path.join(STORAGE_DIR, metric),
+        # TODO(c.chary): implement that in carbonate
+        # start=start, end=end
+    )
+
+
+def _merge_from_staging(staging_dir, metrics_fs, update_cb, chunksize,
+                        start, end):
+    """Merge files in the staging directory with local files."""
+    workers = multiprocessing.Pool()
+
+    # An iterable of pairs of ((staging_dir, start, end), metric)
+    args = staging_dir, start, end
+    args_and_metrics = itertools.izip(itertools.repeat(args), metrics_fs)
+
+    done = 0
+    update_cb(done)
+
+    if not PARALLEL:
+        for i in args_and_metrics:
+            _heal(i)
+        return
+
+    for _ in workers.imap_unordered(
+            _heal, args_and_metrics, chunksize=chunksize):
+        done += 1
+        update_cb(done)
+    workers.close()
+    workers.join()
+
+
+def _get_nodes(config, cluster_name, node=None):
+    """Get the list of nodes in another cluster, excludes 'node'."""
+    res = set()
+    for entry in util.parseDestinations(config.destinations(cluster_name)):
+        res.add(entry[0])  # Only keep the host.
+    if node in res:
+        res.remove(node)
+    return res
+
+
+def _parse_args():
+    """Parse command line arguments."""
+    parser = carbonate_util.common_parser(description='carbonate-sync')
+    parser.add_argument(
+        '-S', '--remote_cluster',
+        default=None,
+        help='Remote cluster name')
+
+    parser.add_argument(
+        '-n', '--node',
+        default=socket.gethostname(),
+        help='Name of the local node.')
+
+    parser.add_argument(
+        '-b', '--batch-size',
+        default=1000,
+        help='Batch size for the merge job.')
+
+    parser.add_argument(
+        '--ssh-options',
+        default=' '.join(SSH_OPTIONS),
+        help='Pass option(s) to ssh. Make sure to use ' +
+        '"--ssh-options=" if option starts with \'-\'')
+
+    parser.add_argument(
+        '--rsync-options',
+        default=' '.join(RSYNC_OPTIONS),
```

Comment:
while for carbon paths there isn't a clear separator, for --ssh_options you can just copy-paste whaterver you would use and quote it. I think it's easier to use this way.